# IMNA
###  1. Introduction
> **`IMNA`** is a fast and effective tool to generate linkage disequilibrium (LD) heatmap from VCF files. It is more time and memory saving than other current tools. LDBlockShow can generate the plots of LD heatmap and interested statistics or annotation results simultaneously. In addition, it also supports subgroup analysis.
<b>IMNA</b> is a fast and effective tool to generate linkage disequilibrium (LD) heatmap from VCF files. It is more time and memory saving than other current tools. LDBlockShow can generate the plots of LD heatmap and interested statistics or annotation results simultaneously. In addition, it also supports subgroup analysis.

###  2. Download and Configure
------------
</br><b>2.2 Download</b>
<pre>
	git clone https://github.com/xjtugenetics/IMNA.git
</pre>

</br><b>2.3 Configure</b>
<pre>
	export IMNA_tk=/path/to/IMNA
</pre>


###  3. Usage
------------
</br><b>3.1 Export gene module </b>

```php
1-Export_module.py
		
        Usage: python3 $IMNA_tk/script/1-Export_module.py <gene module directory> <output prefix>

```
</br> Details for parameters:
<pre>
gene module directory          Directory of all gene module
output prefix                  Prefix of output file
</pre>

</br><b>3.1.2 Other parameters</b>
```php
./bin/LDBlockShow -h
		 
		 Para [-i] is show for [-InVCF], Para [-o] is show for [-OutPut], Para [-r] is show for [-Region]

		-InGenotype    <str>     InPut SNP Genotype Format
		-InPlink       <str>     InPut Plink [bed+bim+fam] or [ped+map] file prefix
		
		-MAF           <float>   Min minor allele frequency filter [0.05]
		-Het           <float>   Max ratio of het allele filter [0.90]
		-Miss          <float>   Max ratio of miss allele filter [0.25]

		-TagSNPCut     <float>  'Strong LD' cutoff for TagSNP [0.80]
		-OutPng                  convert svg 2 png file
		-OutPdf                  convert svg 2 pdf file

```
</br> Details for above parameters:
<pre>
-InGenotype   Input file in genotype format.
-InPlink      The prefix of input file in PLINK format.
-MAF          Filter SNPs with low minor allele frequency (default ≤0.05)
-Het          Filter SNPs with high heterozygosis ratio (default ≥0.9)
-Miss         Filter SNPs with high missing rate (default ≥0.25)

-TagSNPCut    The LD cutoff for selecting tag SNPs. Default is 0.8.
-OutPng       Convert the SVG file to PNG file
-OutPdf       Convert SVG file to Pdf file.
</pre>
<b>Note:</b> If users failed to open small SVG files, please use the "-Outpdf" option to use the PDF file. For large SVG files, "-OutPng" can be used to get a relatively small figure file.

</br><b>3.2 ShowLDSVG</b>
</br>This program is designed for users to optimize the figure (e.g., change colors) generated by LDBlockShow.
</br><b>3.2.1 Brief parameters</b>
```php
./bin/ShowLDSVG
		Options
		 -InPreFix    <s> :  InPut Region LD Result Frefix
		 -OutPut      <s> :  OutPut svg file result

		 -help            :  Show more help with more parameter
```
<pre>
-InPreFix     The prefix of input file (i.e., the output file of LDBlockShow)
-OutPut       The out file (svg, png and pdf format plot files)
-help         More parameters in detail
</pre>

</br><b>3.2.2 Detail parameters</b>
```php
./bin/ShowLDSVG   -h
		 -InGWAS      <s>     : InPut GWAS Pvalue File(chr site Pvalue)
		 -NoLogP              : Do not get the log Pvalue
		 -Cutline     <s>     : show the cut off line of Pvalue
		 -TopSite     <n>     : InPut the Special Site as the peak site(chr:pos)
		 -PointSize   <n>     : set the GWAS point size number
		 -SpeSNPName  <s>     : In File for Special SNP Name(chr site Name)
		 -ShowGWASSpeSNP      : show Special SNP Name in GWAS plot with [-SpeSNPName]

		 -InGFF       <s>     : InPut GFF3 file to show Gene CDS and name
		 -NoGeneName          : No show Gene name,only show stuct
		 -crGene      <s>     : InColor for Gene Stuct [CDS:Intron:UTR:Intergenic]
		                        default: [#e7298a:lightblue:#7570b3:#a6cee3]

		 -crBegin     <s>     : In Start Color RGB [255,255,255]
		 -crMiddle    <s>     : In Middle Color RGB [240,235,75]
		 -crEnd       <s>     : In End Color RGB [255,0,0]
		 -NumGradien  <s>     : In Number of gradien of color
		 -crTagSNP    <s>     : Color for TagSNP [31,120,180]

		 -CrGrid      <s>     : the color of grid edge [white]
		 -WidthGrid   <s>     : the edge-width of gird [1]
		 -NoGrid              : No Show the gird edge

		 -ShowNum             : Show the R^2/D' in the heatmap
		 -NoShowLDist   <n>   : NoShow long physical distance pairwise[1000000]
		 -MerMinSNPNum  <s>   : merge color grids when SNPnumber over N[50]

		 -OutPng              : convert svg 2 png file
		 -OutPdf              : convert svg 2 pdf file
		 -ResizeH             : resize image height; Width be resize in ratio[4096]
```
</br> Details for above parameters:
<pre>
-InGWAS         The statistics file (e.g., association statistics, but other values such as Tajima’s D can also be accepted) for generate plot together with the LD plot. File formatted as: [chr position Pvalue]
-NoLogP         By default, the P value from the -InGWAS file will be -log10 transformed, with this option, the P value will not be transformed.
-Cutline        The significance cutline of the -InGWAS file.
-TopSite        Users can use this option to assign one interested SNP in the GWAS plot (Default is the most significant SNP, can be changed with chr:pos).
-PointSize      Users can use this option (any number over 0) to set the point size.

-InGFF          The GFF file for genomic region annotation. By default, the gene name will be shown in the plot;
-NoGeneName     Gene name will not be shown in the plot with this option.
-SpeSNPName     With this option, users can input a file to indicate the names for interested SNPs, these names will be shown in the heatmap.
-ShowGWASSpeSNP Users can use this option together with the file assigned by '-SpeSNPName' to show the names of interested SNPs in GWAS plot.
-crGene         Define the colors of different genomic regions. By default, CDS, intron, UTR and intergenic regions will be shown in #e7298a, light blue, #7570b3, and #a6cee3, respectively.


Parameters to optimize the color of the heatmap:
-crBegin        Color for no LD (R^2/D’=0) default: white
-crMiddle       Color for R^2/D’=0.5, default: yellow
-crEnd          Color for complete LD (R^2/D’=1), default: red
-NumGradien     The number of gradients from crBegin to crEnd
-crTagSNP       Color for the tag SNP.

Parameters to optimize the grids in the heatmap:
-CrGrid         Border color of the grids, default: white
-WidthGrid      The width of the border, default = 1
-NoGrid         No border

-ShowNum        Show the LD measurement value in the grids (not recommended when SNP number is over 50).
-NoShowLDist    When the distance between SNPs over this number, their pairwise LD will not be showed in the figure. Default is 10,000,000.
-MerMinSNPNum   When number of SNPs over the default 50, ShowLDSVG will merge adjacent same color grids. User can change this number to any integer numbers.
-OutPng         Convert the SVG file to PNG file
-OutPdf         Convert SVG file to Pdf file.
-ResizeH        Set the height of the image (default 4096), which can be used to adjust the resolution for PNG file. The width will be adjusted automatically.
</pre>

</br><b>Note:</b> When SNP number is large (e.g., over 100), the output SVG file might be very large. ShowLDSVG will merge adjacent same color grids. With smaller number of gradients (set by -NumGradien), the figure will be compressed to be smaller. -MerMinSNPNum is used to set the minimum number of SNPs, that’s, if there is more SNPs than this number (default 50), the output SVG will be compressed.

</br><b>3.3 Output files</b>
<pre>
out.site.gz: Remained SNPs after filtering [chr site]
out.blocks.gz: Block file [chr start end block_length SNP_number SNPs]
out.TriangleV.gz: Region Pairwise R^2/D'
out.svg: Output plot in SVG format
out.png: Output plot in png format
out.pdf: Output plot in pdf format
</pre>


###  4. Example
------------

</br>See more detailed usage in the&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <b>[Chinese Documentation](https://github.com/BGI-shenzhen/LDBlockShow/blob/master/LDBlockShow_Manual_Chinese.pdf)</b>
</br>See more detailed usage in the&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; <b>[English Documentation](https://github.com/BGI-shenzhen/LDBlockShow/blob/master/LDBlockShow_Manual_English.pdf)</b>
</br>See the example directory and  Manual.pdf for more detail.


* Example 1)  show Figure with Defaut LD Blocks

<pre>
#../../bin/LDBlockShow   -InVCF Test.vcf.gz   -OutPut  out   -Region  chr11:24100000:24200000  -OutPng -SeleVar 1
../../bin/LDBlockShow   -InVCF Test.vcf.gz   -OutPut  out   -Region  chr11:24100000:24200000  -OutPng -SeleVar 2
# [-SeleVar 1] is D',[-SeleVar 2] is RR ,[-SeleVar 3] are RR and D',[-SeleVar 4] are D' and RR # the default is D'
</pre>

![out.png](https://github.com/BGI-shenzhen/LDBlockShow/blob/master/example/Example1/out.png)


###  5. License
------------
> This software is distributed under the terms of GPL 2.0
</br>

###  6. Source
------------
> [https://github.com/xjtugenetics/IMNA](https://github.com/xjtugenetics/IMNA)
</br>

###  7. Contact
------------
#### Author
> **Yi-Xiao Chen**, **Shan-Shan Dong**, **Yan Guo**, **Tie-Lin Yang**  
> Key Laboratory of Biomedical Information & Genetics Center, School of Life Science and Technology, Xi'an Jiaotong University, Xi'an, Shaanxi Province, 710049, P. R. China  
> [:email:](yangtielin@mail.xjtu.edu.cn) yangtielin@mail.xjtu.edu.cn  
</br>

###  8. Maintainer
> **Yi-Xiao Chen**  
> You can contact [:email:](aaa@stu.xjtu.edu.cn) aaa@stu.xjtu.edu.cn
  when you have any questions, suggestions, comments, etc.
> Please describe in details, and attach your command line and log messages if possible.  
<br>

###  9. Requiremnets
------------
- Python ( >= 3.2 )
- **Python** \( recommended: python2.7 \)
	- numpy( >=1.10.4)
	- mpmath( >=0.19)
	- pandas( >=0.18.0)
	- scipy( >= 0.17.0)
	- rpy2( >= 2.9.0)
- **R** \( >= 3.3.2 \)
</br>



###################### No pain, no gain. #############################

